packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -261,6 +266,23 @@ class NetworkController extends DisposableController
     return enabled;
   }
 
+  Future<bool> recordingSocketTraffic() async {
+    bool enabled = true;
+    final service = serviceConnection.serviceManager.service!;
+    await service.forEachIsolate(
+      (isolate) async {
+        final socketFuture = service.socketProfilingEnabled(isolate.id!);
+        // The above call won't complete immediately if the isolate is paused,
+        // so give up waiting after 500ms.
+        final state = await timeout(socketFuture, 500);
+        if (state?.enabled != true) {
+          enabled = false;
+        }
+      },
+    );
+    return enabled;
+  }
```

Comment:
this is duplicated above, can you add a helper method? 
```
enum NetworkTrafficType {
  http,
  socket,
}

Future<bool> recordingNetworkTraffic({required NetworkTrafficType type}) {
  late final future = switch (type) {
    NetworkTrafficType.http => service.httpEnableTimelineLoggingWrapper(isolate.id!),
    NetworkTrafficType.socket => service.socketProfilingEnabled(isolate.id!),
  };
  ...
}
```

Reply:
Done.